beacon-chain/sync/service.go

```diff
@@ -126,6 +130,37 @@ func (ss *Service) ReceiveBlockHash(data *pb.BeaconBlockHashAnnounce, peer p2p.P
 	return nil
 }
 
+// verifyAttestation verifies incoming attestation with the block, when a block is produced,
+// it is broadcasted at the network layer along with the attestation from its proposer.
+func (ss *Service) verifyAttestation(parentHash [32]byte, attestation *types.Attestation) error {
```

Comment:
This logically makes more sense as a `types.Attestation` method.

Reply:
and the readability will improve if dependencies like `crystallizedState` are defined as a parameter instead of inside the function.